docs/architecture/sdk/internal/index.md

```diff
@@ -0,0 +1,239 @@
+underlying implementation to be internal to the crate with only the public API exposed through the
+`Client` struct. Below is an example of a generator extension for the `Client` struct.
+
+```rust
+/// Generator extension for the Client struct
+#[cfg_attr(feature = "wasm", wasm_bindgen)]
+pub struct GeneratorClient {
+    client: Client,
+}
+
+#[cfg_attr(feature = "wasm", wasm_bindgen)]
+impl GeneratorClient {
+    fn new(client: Client) -> Self {
+        Self { client }
+    }
+
+    /// Generates a password based on the provided request.
+    pub fn password(&self, input: PasswordGeneratorRequest) -> Result<String, PasswordError> {
+        password(input)
+    }
+}
+
+/// Extension which exposes `generator` method on the `Client` struct.
+pub trait GeneratorClientExt {
+    fn generator(&self) -> GeneratorClient;
+}
+
+impl GeneratorClientExt for Client {
+    fn generator(&self) -> GeneratorClient {
+        GeneratorClient::new(self.clone())
+    }
+}
+```
+
+## Language bindings
+
+The SDK is currently exposed with multiple language bindings. Currently we utilize a mix of hand
+written bindings for a C API, and programmatically generated bindings.
+
+### Mobile bindings
+
+We use [UniFFI](https://github.com/mozilla/uniffi-rs/) to generate bindings for the mobile
+platforms, more specifically we publish Android and iOS libraries with Kotlin and Swift bindings,
+respectively. While UniFFI supports additional languages they typically lag a few releases behind
+the UniFFI core library.
+
+The Android bindings are currently published on
+[GitHub Packages](https://github.com/bitwarden/sdk/packages/1945788) in the SDK repository. The
+swift package is published in the [`sdk-swift` repository](https://github.com/bitwarden/sdk-swift).
+
+### Web bindings
+
+For the web bindings we use [`wasm-bindgen`](https://github.com/rustwasm/wasm-bindgen) to generate a
+WebAssembly module that can be used in JavaScript / TypeScript. To ensure compatibility with
+browsers that do not support WebAssembly, we also generate a JavaScript module from the WebAssembly
+that can be used as a fallback.
+
+The WebAssembly module is published on [npm](https://www.npmjs.com/package/@bitwarden/sdk-internal).
+
+### C bindings
+
+Many language bindings utilize the `bitwarden-c` crate that exposes a C API. This is then combined
+with hand written bindings for the specific language. Since manually writing FFI bindings is time
+consuming and difficult we generally provide a JSON based API through the `bitwarden-json` crate
+which allows the language bindings to just contain three FFI functions, `init`, `run_command` and
+`free_memory`.
+
+## Adding New Functionality
+
+Considering adding to or moving code into the SDK? Review these questions to help come to a
+decision.
+
+- Does the functionality or service depend on other functionality that is not currently part of the
+  SDK?
+  - Moving that functionality to the SDK is not recommended at this time.
```

Comment:
suggestion: maybe add a point here to ask/tell the team owning that functionality that they would like it moved to the sdk